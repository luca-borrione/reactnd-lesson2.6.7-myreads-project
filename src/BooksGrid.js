
import React from 'react'
import Book from './Book'
import PropTypes from 'prop-types'

class BooksGrid extends React.Component {

	static propTypes = {
		bookIDs: PropTypes.arrayOf(
					 PropTypes.string.isRequired
				 ).isRequired,
		shelves: PropTypes.arrayOf(PropTypes.string).isRequired,
		updateShelf: PropTypes.func.isRequired
	};

	render() {
		const { bookIDs, shelves, updateShelf } = this.props;

		return (
			<ol className="books-grid">
				{bookIDs.map( (bookID, index) => (

					<Book key={index}
						id={bookID}
						shelves={shelves}
						updateShelf={updateShelf} />

				))}
			</ol>
		);
	}
}


export default BooksGrid
