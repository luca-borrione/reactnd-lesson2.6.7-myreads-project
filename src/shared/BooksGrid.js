import React from 'react';
import Book from './Book';
import PropTypes from 'prop-types';
import { TBook } from './types';
import { diff as ArrayDiff} from '../utils/ArrayUtils';

/**
 * @module
 * @name BooksGrid
 * @extends React.Component
 * @description
 * Prints a list of books.
 * This module is used both in MyReads to show the books for each shelf,
 * and in the SearchPage to show the books retrieved by a search.
 */
class BooksGrid extends React.Component {

	/**
	 * @member
	 * @name propTypes
	 * @description Intended types of the props passed to the component
	 * @property {TBook[]} books - List of the books to show
	 * @property {function} updateBookShelf - see [updateBookShelf]{@link module:App~updateBookShelf}
	 * @static
	 */
	static propTypes = {
		books: PropTypes.arrayOf(
			PropTypes.shape(TBook).isRequired
		).isRequired,
		updateBookShelf: PropTypes.func.isRequired
	};

	/**
	 * @method
	 * @name shouldComponentUpdate
	 * @param {Object} nextProps - ignored as the module does not expect props
	 * @param {TBook[]} nextProps.books - foo
	 * @returns {boolean} - false if the books to show don't change
	 * @private
	 */
	shouldComponentUpdate(nextProps) {
		return ArrayDiff(this.props.books, nextProps.books).length > 0;
	}


	/**
	 * @method
	 * @name render
	 * @description
	 * Renders the [Book]{@link module:Book} component for each book in the given books list
	 * @returns {ReactElement}
	 * @private
	 */
	render() {
		const { books, updateBookShelf } = this.props;
		return (
			<ol className="books-grid">
				{books.map( (book, index) => (

					<Book key={index}
						book={book}
						updateBookShelf={updateBookShelf} />

				))}
			</ol>
		);
	}
}


export default BooksGrid;
