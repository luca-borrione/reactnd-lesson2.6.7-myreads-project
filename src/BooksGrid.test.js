import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import BooksGrid from './BooksGrid';

describe('BooksGrid', () => {

	let books = [];
	const updateBookShelf = () => {};


	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<BooksGrid
				books={books}
				updateBookShelf={updateBookShelf} />, div);
		ReactDOM.unmountComponentAtNode(div);
	});


	it('renders correctly', () => {
		const tree = renderer.create(
			<BooksGrid
				books={books}
				updateBookShelf={updateBookShelf} />
		)
		.toJSON();

		expect(tree).toMatchSnapshot();
	});

});
