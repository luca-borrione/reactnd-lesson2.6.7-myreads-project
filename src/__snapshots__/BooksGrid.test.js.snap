// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksGrid renders correctly 1`] = `
<ol
  className="books-grid"
/>
`;
